### DISCUSSION TITLE
Lastest version on low end device (Openwrt Router)

### DISCUSSION THREAD

**Discussion by hoang-rio** [aio-libs/aiohttp#10026] [Q&A] [answered]
I'm trying to install aiohttp in OpenWrt with mips CPU with command `pip install --only-binary :all: --no-build-isolation aiohttp==3.11.7` but it fail. The reason why I use --only-binary because of my OpenWrt can not install gcc by low disk space.

Current python version on OpenWrt is `3.11.7`. I only can install version 3.6.2 with --only-binary but it did not compability with python `3.11.7`

Any solution for me?

**Comment by webknjaz** [comment, marked as answer]
Looks like we shipped pure-python wheels in the past but not anymore. I hope to restore that soon.
![Screenshot_2024-11-27-02-27-00-18_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://github.com/user-attachments/assets/b08baf3a-7b8b-4620-bd7f-9930bcfb751b)

A workaround would be building with the `AIOHTTP_NO_EXTENSIONS=1` env var set which will disable building the C-extensions.

**Comment by hoang-rio** [comment]
I found a solution by using aiohttp as git submodule and add it's dependencies in requirements.txt and it worked

**Reply by webknjaz** [reply]
I guess, but using the env var would probably be much cleaner: `AIOHTTP_NO_EXTENSIONS=1 FROZENLIST_NO_EXTENSIONS=1 MULTIDICT_NO_EXTENSIONS=1 PROPCACHE_NO_EXTENSIONS=1 YARL_NO_EXTENSIONS=1 pip install aiohttp`. Or you could make the wheel locally via `AIOHTTP_NO_EXTENSIONS=1 FROZENLIST_NO_EXTENSIONS=1 MULTIDICT_NO_EXTENSIONS=1 PROPCACHE_NO_EXTENSIONS=1 YARL_NO_EXTENSIONS=1 pip wheel --wheel-dir=./simple-index aiohttp`, store it somewhere with your deployment artifacts and install from that using `pip install --no-index --find-links=./simple-index aiohttp`: https://packaging.python.org/en/latest/guides/index-mirrors-and-caches/#caching-with-pip.

**Reply by hoang-rio** [reply]
Tried with `AIOHTTP_NO_EXTENSIONS=1 FROZENLIST_NO_EXTENSIONS=1 MULTIDICT_NO_EXTENSIONS=1 PROPCACHE_NO_EXTENSIONS=1 YARL_NO_EXTENSIONS=1 pip install aiohttp` but it make my router crash and auto restart. My router  CPU may can not build anything
